nlpia_bot/clibot.py

```diff
@@ -187,19 +192,20 @@ def reply(self, statement=''):
                     log.error(str(e))
             bot_replies = normalize_replies(bot_replies)
             replies.extend(bot_replies)
+
+        # Weighted random selection of reply from those with top n confidence scores 
```

Comment:
This is a great comment. Code is complicated and comment simplifies it for the reader.